core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -953,6 +962,12 @@ private[log] class Cleaner(val id: Int,
       if(position == startPosition)
         growBuffersOrFail(segment.log, position, maxLogMessageSize, records)
     }
+
+    // check for missing offsets at the end of logSegment
```

Comment:
@junrao  I'm gonna need help explaining why is this happening in live kafka since I'm unaware of origin of probelm when/why and how.

also, I'm gonna remove `if condition` and always updateLatsOffset to expected end of segment as you suggested